src/base.jl

```diff
@@ -234,13 +234,13 @@ function get_components(::Type{T}, sys::ConcreteSystem) where {T <: Component}
     if isconcretetype(T)
         components = get(sys.components, T, nothing)
         if isnothing(components)
-            return Iterators.flatten([[]])
+            return Iterators.flatten([Vector{T}()])
```

Comment:
For readability it would be good to set the type of this function as ComponentIterator{T} - and describe it explicitly in the docstring.  I was very vague the first time around because the return type wasn't consistent.

Reply:
addressed in 197f19e